### DISCUSSION TITLE
Custom Order Number Generation

### DISCUSSION THREAD

**Discussion by nickmonad** [medusajs/medusa#4319] [Q&A]
Is there a way to configure Medusa to use a custom method for order number generation? We ideally would prefer to have large, random numbers for our orders, or at the very least, have them not start from `1`.

**Comment by tsvetann** [comment]
The order numbers Medusa generates for me do not start from 1. Are you referring to something else?

**Reply by nickmonad** [reply]
Interesting, what do they start from for you? I have a few dozen test orders created on a local instance of medusa (`v1.12.0`) and the order numbers start from `1`

![Screenshot_2023-06-15_14-23-44](https://github.com/medusajs/medusa/assets/965744/cf298f01-b907-404c-9177-835ed07c6bce)

**Reply by nickmonad** [reply]
Is it because I'm on a local dev instance? Is there a production config that changes this?

**Reply by tsvetann** [reply]
For me order ids look like this: order_01H0A77YBVCHB563W7JVCJ1784 When you open an order in the admin check the address bar or check in your local DB in the order table

**Reply by nickmonad** [reply]
Well, yes, those order ID formats exist for me as well and how medusa is representing the order in its database, but it's not typical of many stores to give customers order numbers that look like that. You would typically see "Order #716181" or something like that. We would prefer if ours could be a simple number, but not start at 1, for a variety of reasons.

If there was some way to hook into that number generation, that would be great.

**Reply by nickmonad** [reply]
Another issue with giving a customer those `order_` formatted IDs as an "order number", is that it appears shop admins can't search those with the search bar in the admin UI. At least, it doesn't seem to work for me.

**Reply by tsvetann** [reply]
I agree about the issue searching orders by the order id. But what is the use case for searching an order by it's id? If a customer calls you and has questions about an order wouldn't you search the order by customer email for example or customer name?

**Comment by ghost** [comment]
Also interested in this.

**Comment by pevey** [comment]
To make the display number for orders not start at 0, use psql or your favorite GUI db admin app and do:

```sql
SELECT setval('order_display_id_seq', NNNNNNN, true);
```

where NNNNNNN is the integer number you want to use.

**Comment by SGFGOV** [comment]
It would be best if there was a way to customise this via medusa and not
use dB..

Order id's are important from a customer service perspective.. for example
a repeat customer has an issue with the 3rd order.  Or the customer has
placed multiple orders that are yet to be fulfilled or are partially
fulfilled

The order number is the only way to identify the order. Asking a customer
for a 16 digit number is prone to error..

Best Regards
Govind

On Thu, 21 Sept, 2023, 03:26 Lacey Pevey, ***@***.***> wrote:

> To make the display number for orders not start at 0, use psql or your
> favorite GUI db admin app and do:
>
> SELECT setval('order_display_id_seq', NNNNNNN, true);
>
> where NNNNNNN is the integer number you want to use.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/medusajs/medusa/discussions/4319#discussioncomment-7063260>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXEQJHDK3D5SR6CMQXZ4EG3X3NRAVANCNFSM6AAAAAAZIAYCYU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 

This message (including any attachments) may contain confidential, 
proprietary, privileged and/or private
information. The information is 
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the 
sender immediately, and delete the
message and any attachments. Any 
disclosure, reproduction, distribution or other use of this message or
any 
attachments by an individual or entity other than the intended recipient is 
prohibited.

**Comment by picardplaisimond** [comment]
It's 2026. Have we found a solution to this ?

**EDIT:**  Nevermind, here's the solution: https://docs.medusajs.com/resources/commerce-modules/order/custom-display-id
